providers/cliflag/cliflag.go

```diff
@@ -0,0 +1,118 @@
+// Package cliflag implements a koanf.Provider that reads commandline
+// parameters as conf maps using ufafe/cli flag.
+package cliflag
+
+import (
+	"strings"
+
+	"github.com/knadh/koanf/maps"
+	"github.com/urfave/cli/v2"
```

Comment:
As v1, v2 are actively maintained (and widely used), and v3 is about to release, it will be really nice to have support for all the versions urfave/cli versions natively on koanf. But, this PR adds support only for `urfave/cli/v2`.

I am still confused how different versions of the same `provider` (here, `cliflag`) can be structured and maintained. Then I saw your feature request #333. I have added my comments/thoughts in that [issue](https://github.com/knadh/koanf/issues/333#issuecomment-2472288760). Please help me how to structure different versions of `urfave/cli` inside koanf.